opslevel/resource_opslevel_service.go

```diff
@@ -1,360 +1,458 @@
+	updatedServiceResourceModel, diags := NewServiceResourceModel(ctx, *service)
+	updatedServiceResourceModel.LastUpdated = timeLastUpdated()
+	resp.Diagnostics.Append(diags...)
+
+	tflog.Trace(ctx, "updated a service resource")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &updatedServiceResourceModel)...)
+}
+
+func (r *ServiceResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data ServiceResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := r.client.DeleteService(data.Id.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete service, got error: %s", err))
+		return
+	}
+	tflog.Trace(ctx, "deleted a service resource")
+}
+
+func (r *ServiceResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+// Assigns new aliases from terraform config to service and deletes aliases not in config
+func reconcileServiceAliases(client opslevel.Client, aliasesFromConfig []string, service *opslevel.Service) error {
+	// delete service aliases found in service but not listed in Terraform config
+	for _, managedAlias := range service.ManagedAliases {
+		if !slices.Contains(aliasesFromConfig, managedAlias) {
+			if err := client.DeleteServiceAlias(managedAlias); err != nil {
+				return err
+			}
+		}
+	}
+
+	// create aliases listed in Terraform config but not found in service
+	newServiceAliases := []string{}
+	for _, configAlias := range aliasesFromConfig {
+		if !slices.Contains(service.ManagedAliases, configAlias) {
+			newServiceAliases = append(newServiceAliases, configAlias)
+		}
+	}
+	if len(newServiceAliases) > 0 {
+		if _, err := client.CreateAliases(service.Id, newServiceAliases); err != nil {
+			return err
+		}
+	}
+	service.ManagedAliases = aliasesFromConfig
+	return nil
+}
+
+// Assigns new tags from terraform config to service and deletes tags not in config
+func reconcileTags(client opslevel.Client, tagsFromConfig []string, service *opslevel.Service) error {
+	// delete service tags found in service but not listed in Terraform config
+	existingTags := []string{}
+	for _, tag := range service.Tags.Nodes {
+		flattenedTag := flattenTag(tag)
+		existingTags = append(existingTags, flattenedTag)
+		if !slices.Contains(tagsFromConfig, flattenedTag) {
+			if err := client.DeleteTag(tag.Id); err != nil {
+				return err
+			}
+		}
+	}
+
+	// format tags listed in Terraform config but not found in service
+	tagInput := map[string]string{}
+	for _, tag := range tagsFromConfig {
+		parts := strings.Split(tag, ":")
+		if len(parts) != 2 {
+			return fmt.Errorf("[%s] invalid tag, should be in format 'key:value' (only a single colon between the key and value, no spaces or special characters)", tag)
```

Comment:
We can remove this if statement and error altogether by using `tagFormatValidator`

Reply:
This is fine to keep here too since we don't have any guarantees the code between the validation done in `tagFormatValidator` and `reconcileTags` doesn't change things